### PR TITLE
CP-33380: update opam for dune 2 test compatibility

### PR DESCRIPTION
**Pull request by psafont**
Tests were failing because the cli rules are not available when restricting the rules to only a the `message-switch` package.

Also removed tuareg runes from dune files.

### CODE REVIEW

File: core_test/dune [hunk 1]
```diff
@@ -29,9 +20,6 @@ let () = Printf.ksprintf Jbuild_plugin.V1.send {|
     lwt/link_test_main.exe
     ../switch/switch_main.exe
     ../cli/main.exe
-    basic-rpc-test.sh
```
Comment (edwintorok) [inline core_test/dune]: Did you intend to drop this?

Comment (psafont) [reply core_test/dune]: Yes, it's not needed since the dependency is inferred from the action.

File: message-switch.opam [hunk 2]
```diff
@@ -9,7 +9,7 @@ tags: [ "org:xapi-project" ]
 build: [
   ["./configure" "--bindir" "%{bin}%"]
   ["dune" "build" "-p" name "-j" jobs]
-  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+  ["dune" "runtest" "-j" jobs] {with-test}
```
Comment (lippirk) [inline message-switch.opam]: did you make sure that this didn't change the number of message switch tests that run?

Comment (psafont) [reply message-switch.opam]: The only tests run before and after are using the shell script.

**Comment by aantron** [comment]
> Also removed tuareg runes from dune files.

For my reference, what was the motivation for this?

**Comment by psafont** [comment]
Automation around coverage around the toolstack project has been unmaintained for some time. This along the possibility ocaml syntax might be deprecated means that this syntax is being removed.

There's a hope dune will support bisect_ppx natively soon-ish and it will be easy to integrate it.

**Pull request merged**